lib/linux_admin/ssh_agent.rb

```diff
@@ -0,0 +1,85 @@
+require 'net/ssh'
+require 'fileutils'
+module LinuxAdmin
+  class SshAgent
+    attr_reader :pid
+    attr_reader :sock
+
+    def initialize(agent_socket, ssh_private_key)
+      FileUtils.mkdir_p(File.dirname(agent_socket))
+      agent_details = `ssh-agent -a #{agent_socket}`
+      @sock         = agent_details.split("=")[1].split(" ")[0].chop
+      @pid          = agent_details.split("=")[2].split(" ")[0].chop
+      IO.popen({"SSH_AUTH_SOCK" => @sock, "SSH_AGENT_PID" => @pid}, ["ssh-add", "-"], :mode => 'w') do |f|
+        f.puts(ssh_private_key)
+        if $?.to_i != 0
+          raise StandardError, "Couldn't add key to agent"
+        end
+      end
+    end
+
+    def perform_commands(ip, username, commands, _additional_flags = "")
```

Comment:
Why not just remove the `additional_flags` param here? Is there some other API you are trying to adhere to?